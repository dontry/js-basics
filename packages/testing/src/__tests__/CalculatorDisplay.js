import "react-testing-library/cleanup-after-each";
import React from "react";
import { render } from "react-testing-library";
import CalculatorDisplay from "../components/CalculatorDisplay";
import serializer from "jest-emotion";
import * as emotion from "emotion";
//We could actually snapshot container.innerHTML,
//but the problem with that is any change to any of these attributes would negate the entire snapshot
// would be harder to read diffs of the snapshot.
//If you only have one child that you're rendering, then it doesn't actually make any sense to snapshot that diff.
//You could actually snapshot the firstChild which will be the root node of the thing that you're rendering.
//For us that's this div here.

expect.addSnapshotSerializer(serializer);

test("renders with correct styles", () => {
  // const { container } = render(<H1>hello world</H1>);
  const { container } = render(<CalculatorDisplay value="10" />);

  expect(container).toMatchSnapshot();
});
