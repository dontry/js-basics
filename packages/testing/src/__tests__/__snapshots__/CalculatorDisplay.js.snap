// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
.emotion-0 {
  color: white;
  background: #1c191c;
  line-height: 130px;
  font-size: 6em;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

.emotion-0 {
  color: white;
  background: #1c191c;
  line-height: 130px;
  font-size: 6em;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="autoScalingText"
      style="transform: scale(1,1);"
    >
      10
    </div>
  </div>
</div>
`;
